http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -68,52 +127,28 @@
 
   WebSocketConnection(final URI serverUri) {
     this.serverUri = serverUri;
-    client =
-        new WebSocketClient(serverUri) {
-          @Override
-          public void onOpen(final ServerHandshake serverHandshake) {
-            synchronized (queuedMessages) {
-              connectionIsOpen = true;
-              queuedMessages.forEach(this::send);
-              queuedMessages.clear();
-            }
-          }
-
-          @Override
-          public void onMessage(final String message) {
-            listener.messageReceived(message);
-          }
-
-          @Override
-          public void onClose(final int code, final String reason, final boolean remote) {
-            if (remote) {
-              log.severe("Connection to server closed: " + reason);
-            }
-            pingSender.cancel();
-            listener.connectionClosed(reason);
-          }
-
-          @Override
-          public void onError(final Exception exception) {
-            listener.handleError(exception);
-          }
-        };
     pingSender =
         Timers.fixedRateTimer("websocket-ping-sender")
             .period(45, TimeUnit.SECONDS)
             .delay(45, TimeUnit.SECONDS)
             .task(
                 () -> {
-                  if (client.isOpen()) {
-                    client.sendPing();
+                  if (!client.isOutputClosed()) {
+                    client
+                        .sendPing(ByteBuffer.wrap("Ping".getBytes(StandardCharsets.UTF_8)))
+                        .exceptionally(logWebSocketError("Failed to send ping."));
```

Comment:
If the ping cannot be sent, would it maybe mean the connection is already closed?
Would we also want users to submit bug reports whenever this happens?

I suspect we may want to no-op if the ping cannot be sent.

Reply:
Could potentially be the case if we close it between actually sending the ping and the if check, but the exception type will be IOException, so we'll have to check the outputstream again to be sure

Reply:
> Would we also want users to submit bug reports whenever this happens?

My implication with this was that given the current handling, we are offering users to submit bug reports. Severe logging means "the application crashed, it should not have, the error is likely unrecoverable, the user should submit a bug report as the situation should never happen"

We should not be getting bug reports where our response is: "yeah, that's normal, we'll close this now."